### PR TITLE
Update SRS.md

### PR DESCRIPTION
**Pull request by IlyaRozumouski**

### CODE REVIEW

**Review by Maks122525 — APPROVED** [review]

File: SRS.md [hunk 1]
```diff
@@ -1,28 +1,98 @@
...
+    
+  __The list displays following employees' information:__  
+  1. Department - place, where employee is working;  
+  2. Name of the employee;  
+  3. Date of birth of the employee;  
+  4. Employee's salary.  
+    
+  __Filtering by date:__  
+  1. In the employees list view mode, the user sets a date filter and presses the refresh list button (to the right of the date entry field);  
+  2. The application will show the employees only for a certain period of time.
```
Comment (inauros) [inline SRS.md:20]: 1. How to reset a search result?
2. What if we need to find employees who were born after a certain date?
3. What if we need to find employees who were born before a certain date?

File: SRS.md [hunk 2]
```diff
@@ -1,28 +1,98 @@
...
+  __Filtering by date:__  
+  1. In the employees list view mode, the user sets a date filter and presses the refresh list button (to the right of the date entry field);  
+  2. The application will show the employees only for a certain period of time.  
 
-![image](https://user-images.githubusercontent.com/83345134/127283891-5c3f5bcd-b593-4d9d-9a74-4706bfdae2cb.png "Departments list")
-![image](https://user-images.githubusercontent.com/83345134/127284946-62fb3d35-98d4-40a6-8d0a-923d58bc3467.png "Add department")
-![image](https://user-images.githubusercontent.com/83345134/127284991-63ca404a-6d7a-400a-a6e1-c7b6b9560a7e.png "Edit department")
-![image](https://user-images.githubusercontent.com/83345134/127285071-2b699f96-0ac5-4421-b7ab-d78414e6d1d0.png "Delete department")
+  __Restrictions:__  
+  * Start date of the period should be less then end date of the period;
```
Comment (inauros) [inline SRS.md:23]: What happens if the user enters an end date < start date?

File: SRS.md [hunk 3]
```diff
@@ -1,28 +1,98 @@
...
+  * If end date is blank, then filtering by start date only.  
+  * Updating data after selecting the filtering conditions is carried out by pressing the “Refresh” button.  
 
+      
+  ### 2.2. Add employee  
+  __Main scenario:__  
+  * User clicks the “Add” button in the employees list view mode;  
+  * Application displays form to enter employee data;  
+  * User enters employee’s data and presses “Save” button;  
+  * If any data is entered incorrectly, incorrect data messages are displayed;
```
Comment (inauros) [inline SRS.md:34]: It would be great to see a mockup with an error messages.

File: SRS.md [hunk 4]
```diff
@@ -1,28 +1,98 @@
...
 
+      
+  ### 2.2. Add employee  
+  __Main scenario:__  
+  * User clicks the “Add” button in the employees list view mode;  
+  * Application displays form to enter employee data;  
+  * User enters employee’s data and presses “Save” button;  
+  * If any data is entered incorrectly, incorrect data messages are displayed;  
+  * If entered data is valid, then record is adding to database;  
+  * If error occurs, then error message is displaying;
```
Comment (inauros) [inline SRS.md:36]: It would be great to see a mockup with an error messages.

File: SRS.md [hunk 5]
```diff
@@ -1,28 +1,98 @@
...
+  * Name –  maximum length of 45 characters;  
+  * Salary – maximum value is 2000;  
+  * Date of birth – date in format dd/mm/yyyy.   
+      
+  ### 2.3 Edit employee  
+  __Main scenario:__  
+  * User clicks the “Edit” button in the employees list view mode;  
+  * Application displays form to enter employee data;  
+  * User enters cemployee’s data and presses “Save” button;  
+  * If any data is entered incorrectly, incorrect data messages are displayed;
```
Comment (inauros) [inline SRS.md:66]: It would be great to see a mockup with an error messages.

File: SRS.md [hunk 6]
```diff
@@ -1,28 +1,98 @@
...
+  * Date of birth – date in format dd/mm/yyyy.   
+      
+  ### 2.3 Edit employee  
+  __Main scenario:__  
+  * User clicks the “Edit” button in the employees list view mode;  
+  * Application displays form to enter employee data;  
+  * User enters cemployee’s data and presses “Save” button;  
+  * If any data is entered incorrectly, incorrect data messages are displayed;  
+  * If entered data is valid, then edited data is added to database;  
+  * If error occurs, then error message is displaying;
```
Comment (inauros) [inline SRS.md:68]: It would be great to see a mockup with an error messages.

File: SRS.md [hunk 7]
```diff
@@ -1,28 +1,98 @@
...
+![image](https://user-images.githubusercontent.com/83345134/127285373-3f14cb74-9510-4dc3-8e3a-2c2c01dee4a4.png "Edit employee")  
+    Pic. 2.3 Edit employee.    
+    
+  ### 2.4 Removing employee  
+  __Main scenario:__  
+  * The user, while in the list of employees mode, presses the "Delete" button in the selected employee line;  
+  * Application displays confirmation dialog “Please confirm delete employee?”;  
+  * The user confirms the removal of the employee;  
+  * Record is deleted from database;  
+  * If error occurs, then error message displays;
```
Comment (inauros) [inline SRS.md:87]: It would be great to see a mockup with an error messages.

File: SRS.md [hunk 8]
```diff
@@ -1,28 +1,98 @@
...
+  __Main scenario:__  
+  * User clicks the “Add” button in the employees list view mode;  
+  * Application displays form to enter employee data;  
+  * User enters employee’s data and presses “Save” button;  
+  * If any data is entered incorrectly, incorrect data messages are displayed;  
+  * If entered data is valid, then record is adding to database;  
+  * If error occurs, then error message is displaying;  
+  * If new employee record is successfully added, then list of employees with added records is displaying.  
+ 
+  __Cancel operation scenario:__
```
Comment (inauros) [inline SRS.md:39]: I think this scenario should be described in a more general way, as it applies to all actions: "Add", "Edit" and "Delete".

File: SRS.md [hunk 9]
```diff
@@ -1,28 +1,98 @@
...
+  __Main scenario:__  
+  * User clicks the “Edit” button in the employees list view mode;  
+  * Application displays form to enter employee data;  
+  * User enters cemployee’s data and presses “Save” button;  
+  * If any data is entered incorrectly, incorrect data messages are displayed;  
+  * If entered data is valid, then edited data is added to database;  
+  * If error occurs, then error message is displaying;  
+  * If employee’s record is successfully edited, then list of employees with added records is displaying.  
+ 
+  __Cancel operation scenario:__
```
Comment (inauros) [inline SRS.md:71]: I think this scenario should be described in a more general way, as it applies to all actions: "Add", "Edit" and "Delete".

File: SRS.md [hunk 10]
```diff
@@ -1,28 +1,98 @@
...
+  ### 2.4 Removing employee  
+  __Main scenario:__  
+  * The user, while in the list of employees mode, presses the "Delete" button in the selected employee line;  
+  * Application displays confirmation dialog “Please confirm delete employee?”;  
+  * The user confirms the removal of the employee;  
+  * Record is deleted from database;  
+  * If error occurs, then error message displays;  
+  * If employee record is successfully deleted, then list of employees without deleted records is displaying.  
+      
+  __Cancel operation scenario:__
```
Comment (inauros) [inline SRS.md:90]: I think this scenario should be described in a more general way, as it applies to all actions: "Add", "Edit" and "Delete".

**Review by inauros — CHANGES_REQUESTED** [review]
1. In accordance with the task, a folder "documentation" should be created and SRS should be placed in it.
2. Requirement from the template: display a list of employees in departments - this means that users should be able to select, search, or filter employees associated with a specific department in some way.
3. Why do you create a separate branch for each feature?

File: SRS.md (hunk 8 above)
Comment (IlyaRozumouski) [reply SRS.md:39]: I cant get it, the scenarios of editing, adding or deleting are described by every move user can do. They are separated by each other subtitle

Comment (inauros) [reply SRS.md:39]: @IlyaRozumouski  - I'm talking about the "Cancel" scenario. You can combine 3 cases (Add, Edit, Delete) into just one and describe the "Cancel" action in a more general way.

**Comment by IlyaRozumouski** [comment]
Requirement from the template: display a list of employees in departments - this means that users should be able to select, search, or filter employees associated with a specific department in some way.
@inauros, you mean to add another filter for filtering employees by each department?

File: SRS.md (hunk 1 above)
Comment (IlyaRozumouski) [reply SRS.md]: fixed

File: SRS.md (hunk 2 above)
Comment (IlyaRozumouski) [reply SRS.md]: fised

File: SRS.md (hunk 3 above)
Comment (IlyaRozumouski) [reply SRS.md]: mockups are ready

File: SRS.md (hunk 8 above)
Comment (IlyaRozumouski) [reply SRS.md]: The extra subtitle with the general moves to cancel operation is here

**Comment by inauros** [comment]
@IlyaRozumouski 

> @inauros, you mean to add another filter for filtering employees by each department?
Yes, this is what I mean.

> In accordance with the task, a folder "documentation" should be created and SRS should be placed in it.
Not fixed yet.

> Why do you create a separate branch for each feature?
The question was not answered.

**Comment by IlyaRozumouski** [comment]
> > Why do you create a separate branch for each feature?
> 
> The question was not answered.

I decided to create new branch to exclude conflicting files. When i create new file, github asks me where it shold save file. I have two ways: save into master branch (but its wrong way, file may contain mistakes) or save it to another branch and pull request. Thats why i do so

**Review by inauros — APPROVED** [review]

**Review by KisKass — APPROVED** [review]

**Pull request merged**